### DISCUSSION TITLE
Unable to login with SSO

### DISCUSSION THREAD

**Discussion by alecor191** [langfuse/langfuse#2142] [Support] [answered]
I followed instructions ([langfuse docs](https://langfuse.com/docs/deployment/self-host#sso) and [NextAuth.js docs](https://next-auth.js.org/providers/azure-ad)), to configure SSO with AzureAD.

I also tried to set different permutations of `AUTH_DOMAINS_WITH_SSO_ENFORCEMENT`, `AUTH_DISABLE_USERNAME_PASSWORD`, and `AUTH_DISABLE_SIGNUP`.

However, I'm not able to login via SSO. Specifically, I always see the username/password login screen. I was expecting some home realm discovery (entering the email address will change the UI to redirect to the SSO IdP, or a button that says "login with SSO").

So when I enter the username and some random password, the UI indicates that this user is set up for SSO and I should "Please use SSO" ...but how do I "use SSO" as a user of the website? I don't understand how I can tell the UI to log me in via SSO and not show the password field.

![image](https://github.com/langfuse/langfuse/assets/6270008/f76d5f36-91ac-4130-a677-dbe52ffd4ab8)

And if I set `AUTH_DISABLE_SIGNUP` to `true`, then I see the following screen (clicking on `Sign in` does nothing):

![image](https://github.com/langfuse/langfuse/assets/6270008/d0db8790-7050-49b5-925b-d7b5dca3e1e4)

It would be great if anyone has any pointers, as I'm sure I'm missing something obvious. Thanks!

**Comment by alecor191** [comment, marked as answer]
Resolved: The issue was that I followed the instructions in the [NextAuth.js docs](https://next-auth.js.org/providers/azure-ad) as it was referenced from the [langfuse docs](https://langfuse.com/docs/deployment/self-host#sso). However, the name of the env vars is different. I had to use the env var name from [langfuse docs](https://langfuse.com/docs/deployment/self-host#sso) when following instructions in [NextAuth.js docs](https://next-auth.js.org/providers/azure-ad).

**Reply by marcklingen** [reply]
perfect, thanks for confirming that it works as expected now. We just added a reference to the nextauth docs in case there are further configuration options that are relevant to check before adding the oauth app.
